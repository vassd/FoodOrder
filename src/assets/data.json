{
  "restaurant": {
    "name": "3rd rest"
  },
  "menu": {
    "items": [
      {
        "name": "Pizza Margherita",
        "price": 7.00,
        "currency": "RON",
        "ingredients": [
          "tomato sauce",
          "cheese"
        ],
        "category": 0
      },
      {
        "name": "Pizza Prosciutto",
        "price": 10.00,
        "currency": "RON",
        "ingredients": [
          "tomato sauce",
          "cheese",
          "ham"
        ],
        "category": 0
      },
      {
        "name": "Spaghetti Bolognese",
        "price": 9.00,
        "currency": "RON",
        "ingredients": [
        ],
        "category": 1
      },
      {
        "name": "Spaghetti Frutti di Mare",
        "price": 12.00,
        "currency": "RON",
        "ingredients": [
          "with sea food"
        ],
        "category": 1
      }
    ],
    "categories": [
      {
        "id": 0,
        "name": "Pizza"
      },
      {
        "id": 1,
        "name": "Pasta"
      }
    ]
  }
}