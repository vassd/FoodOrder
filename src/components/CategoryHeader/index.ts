export { CategoryHeader } from './CategoryHeader'
export type { Category } from './CategoryHeader'
