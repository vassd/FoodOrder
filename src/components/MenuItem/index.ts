export { MenuItem } from './MenuItem'
export type { MenuItemInterface } from './MenuItem'
