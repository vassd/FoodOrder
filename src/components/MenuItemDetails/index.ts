export { MenuItemDetails } from './MenuItemDetails'
export type { Topping, Crust } from './MenuItemDetails'
