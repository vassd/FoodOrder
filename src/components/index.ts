export { MenuHeader } from './MenuHeader'
export { CategoryHeader } from './CategoryHeader'
export type { Category } from './CategoryHeader'
export { MenuItems } from './MenuItems'
export { MenuItem } from './MenuItem'
export type { MenuItemInterface } from './MenuItem'
