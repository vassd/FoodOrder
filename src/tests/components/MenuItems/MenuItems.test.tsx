import React from 'react'
import '@testing-library/jest-dom'
import { describe, expect, test } from 'vitest'
import { render } from '@testing-library/react'
import { MenuItems } from 'components'
import data from 'assets/data.json'

describe('MenuItems test', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <MenuItems items={data.menu.items} categories={data.menu.categories} />
    )

    expect(container).toMatchSnapshot()
  })
})
