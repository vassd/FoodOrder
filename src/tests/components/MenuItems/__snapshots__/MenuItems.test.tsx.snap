// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MenuItems test > should match snapshot 1`] = `
<div>
  <div
    class="_category-header_d36887"
  >
    Pizza
  </div>
  <ul
    class="MuiList-root css-1mk9mw3-MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container _menu-item_9fa3c5 css-1vam7s3-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 _menu-item__name-ingredients-list_9fa3c5 css-1qpzhgb-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-b3ynvt-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <strong>
                Pizza Margherita
              </strong>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              (tomato sauce, cheese)
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 _menu-item__price_9fa3c5 css-1equabv-MuiGrid-root"
        >
          7 RON
        </div>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container _menu-item_9fa3c5 css-1vam7s3-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 _menu-item__name-ingredients-list_9fa3c5 css-1qpzhgb-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-b3ynvt-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <strong>
                Pizza Prosciutto
              </strong>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              (tomato sauce, cheese, ham)
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 _menu-item__price_9fa3c5 css-1equabv-MuiGrid-root"
        >
          10 RON
        </div>
      </div>
    </li>
  </ul>
  <div
    class="_category-header_d36887"
  >
    Pasta
  </div>
  <ul
    class="MuiList-root css-1mk9mw3-MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container _menu-item_9fa3c5 css-1vam7s3-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 _menu-item__name-ingredients-list_9fa3c5 css-1qpzhgb-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-b3ynvt-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <strong>
                Spaghetti Bolognese
              </strong>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 _menu-item__price_9fa3c5 css-1equabv-MuiGrid-root"
        >
          9 RON
        </div>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container _menu-item_9fa3c5 css-1vam7s3-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 _menu-item__name-ingredients-list_9fa3c5 css-1qpzhgb-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-b3ynvt-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <strong>
                Spaghetti Frutti di Mare
              </strong>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              (with sea food)
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 _menu-item__price_9fa3c5 css-1equabv-MuiGrid-root"
        >
          12 RON
        </div>
      </div>
    </li>
  </ul>
</div>
`;
